\chapter{Gene Regulatory Networks}\label{grn}
\lhead[\fancyplain{}{\bfseries\thepage}]{\fancyplain{}{\bfseries\rightmark}}


\section{Definition}

Gene regulation controls the expression of genes and, consequently, all cellular functions. Gene expression is a process that involves transcription of the gene into mRNA, followed by translation to a protein, which may be subject to post-translational modification \cite{K40}. The transcription process is controlled by transcription factors (TFs) that can work as activators or inhibitors. TFs are themselves encoded by genes and subject to regulation, which altogether forms complex regulatory networks. 
\begin{figure}
\centering
\includegraphics{GRN.png}
\end{figure}
Cells efficiently carry out molecular synthesis, energy transduction, and signal processing across a range of environmental conditions by networks of genes, which we define broadly as networks of interacting genes, proteins, and metabolites \cite{K37}. Formally speaking, a gene regulatory network or genetic regulatory network (GRN) is a collection of DNA segments in a cell which interact with each other (indirectly through their RNA and protein expression products) and with other substances in the cell, thereby governing the rates at which genes in the network are transcribed into mRNA. In general, each mRNA molecule goes on to make a specific protein (or set of proteins). In some cases this protein will be structural, and will accumulate at the cell-wall or within the cell to give it particular structural properties. 



\section{Role within the cell}

Gene Regulatory Networks (GRNs) control biological process of all organisms. The complex control systems underlying development have probably been evolving for more than a billion years. They regulare the expression of thousand of genes in any given biological process. They are essentially hardwired genomic regulatory codes, the role of which is to speicify the sets of genes that must be expressed in specific spatial and temporal patterns. In physical terms, these control system consist of many thousands of modular DNA sequences. Each module receives and integrates multiple inputs, in the form of regulatory proteins (activators and repressors) that recognize specific sequences within them. The end result is the precise transcriptional control of the associated genes. 
Functional linkages between these particular genes, and their associated regulatory modules, define the core networks underlying development. They explain exactly how genomic sequence encodes the regulation of expression of the sets of genes that generate patterns and execute the construction of multiple states of differentiation.

The regulatory genome is a logic processing system: every regulatory module contained in the genome receives multiple inputs and processes in ways that can be mathematically represented as combinations of logic functions. 

Definitive regulatory functions emerge only from the architecture of intergenic linkages, and these functions are not visible at the level of any individual genes. So gene regulatory networks can be determined only by experimental molecular biology in which the functional meaning of given regulatory sequences is directly determined.

GRNs have a complex structure: they are inhomogeneous compositions of different kinds of subnetworks, each performing a specific kind of function. Some subnetworks are used in many processes.
\section{Bo}
In recent years, single-cell-resolution measurements have revealed unprecedented levels of cell-to-cell heterogeneity within tissues. The discovery of this ever-present heterogeneity is driving a more nuanced view of cell phenotype, wherein cells exist along a continuum of cell-states, rather than conforming to discrete classifications. The comprehensive view of diverse cell states revealed by single cell measurements is also affording new opportunities to discover molecular regulators of cell phenotype and dynamics of lineage commitment (Trapnell et al., 2014; Olsson et al., 2016; Briggs et al., 2018). For example, single cell transcriptomics have revealed the widespread nature of multilineage priming (MLP), a phenomenon wherein individual, multipotent cells exhibit “promiscuous” coexpression of genes associated with distinct lineages prior to commitment (Nimmo et al., 2015). In principle, mathematical modeling of gene regulatory network dynamics can provide a theoretical foundation for understanding cell heterogeneity and gene expression dynamics, by quantitatively linking molecular-level regulatory mechanisms with observed cell states. However, due to the molecular complexity of gene regulatory mechanisms, it remains challenging to integrate such models with single-cell data.

Mathematical models of gene regulatory network dynamics can account for (and at least partially reproduce) observed cellular heterogeneity in two primary ways. First, gene network models are multistable dynamical systems, meaning a given network has the potential to reach multiple stable states of gene expression. These states arise from the dynamic interplay of activation, inhibition, feedback, and nonlinearity (Kauffman, 1969; MacArthur et al., 2009; Huang, 2012). Second, some mathematical models inherently treat cellular noise. This noise, or stochasticity, is modeled in various ways depending on assumptions about the source (Peccoud and Ycart, 1995; Arkin et al., 1998; Kepler and Elston, 2001; Swain et al., 2002). Discrete, stochastic models of gene regulation, which track discrete molecular entities, regulatory-protein binding kinetics, and binding states of promoters controlling gene activity, have formed the basis of biophysical theories of gene expression noise due to so-called intrinsic molecular noise (Peccoud and Ycart, 1995; Thattai and van Oudenaarden, 2001; Kepler and Elston, 2001; Pedraza and Paulsson, 2008). Such stochastic gene-regulation mechanisms have also been incorporated into larger regulatory network models using the formalism of stochastic biochemical reaction networks, and have been utilized to explore how molecular fluctuations can cause heterogeneity within phenotype-states and promote stochastic transitions between phenotypes (Feng and Wang, 2012; Sasai et al., 2013; Zhang and Wolynes, 2014; Tse et al., 2015).

The quantitative landscape of cellular states is another concept that is increasingly utilized to describe cellular heterogeneity. Broadly, the cellular potential landscape (first conceptualized by Waddington (Wang et al., 2011; Huang, 2012; Waddington, 2014) is a function in high-dimensional space (over many molecular observables, typically expression levels of different genes), that quantifies the stability of a given cell-state. In analogy to potential energy (gravitational, chemical, electric, etc.), cell states of higher potential are less stable than those of lower potential. The landscape concept inherently accounts for cellular heterogeneity, since it holds that a continuum of states is theoretically accessible to the cell, with low-potential states (in “valleys”) more likely to be observed than high-potential states. The landscape is a rigorously defined function derived from the dynamics of the underlying gene network model, according to some choice of mathematical formalism (Wang et al., 2011; Bhattacharya et al., 2011; Huang, 2012; Zhou et al., 2016). For stochastic gene network models that inherently treat noise, the landscape is directly obtained from the computed probability distribution over cell-states (Cao and Liang, 2008; Micheelsen et al., 2010; Feng and Wang, 2012; Tse et al., 2015).

Stochastic modeling of gene network dynamics has been employed in various forms for analysis of single cell measurements. For example, application of noisy dynamical systems theory has shed light on cell-state transitions (Mojtahedi et al., 2016; Jin et al., 2018; Lin et al., 2018). Stochastic simulations of gene network dynamics have been used to develop and/or benchmark tools for network reconstruction (Schaffter et al., 2011; Dibaeinia and Sinha, 2019; Bonnaffoux et al., 2019) Stochastic model-aided analysis of single-cell measurements has been demonstrated to yield insights on gene regulatory mechanisms (Munsky et al., 2018). However, few existing analysis methods utilize discrete-molecule, stochastic models, which fully account for intrinsic gene expression noise and its impact on cell-state, to aid in the interpretation of noisy distributions recovered from single cell RNA sequencing data. There exists an opportunity to link such biophysical, stochastic models, which reproduce intrinsic noise and cell heterogeneity in silico, to single cell datasets that characterize cell heterogeneity in vivo. In particular, the landscape of heterogeneous cell-states computed from discrete stochastic models can be directly compared to single-cell measurements.


\section{GRN}
The cells of living organisms di erentiate within the developing embryo into
the various cell types that form tissues by a process that is regulated at the
molecular level by DNA sequences, encoding genes that produce proteins that
regulate other genes. All eukaryotic cells in an organism carry an identical set
of genes, some of which are expressed others not. A cell type is de ned by the
particular subset of genes that are expressed. The gene expression pattern of
a cell needs to be stable but also adaptable \cite{K41}.



Genes regulate each other's activity by coding for transcription factors,
which may enhance or repress the expression of other genes by binding (pos-
sibly in combination) at particular sites. Though a particular gene directly
regulates just a small set of other genes, those genes regulate other genes in
turn, so a gene will indirectly in uence the activity of many genes downstream.
Conversely, a particular gene is indirectly in uenced by many genes upstream.
A gene may directly or indirectly contribute to regulating itself. The result is
a genomic regulatory network, a complex feedback web of genes turning each
other on and o . This may be interpreted as an idealized dynamical system of
model genes with directional links (transcription factors), updating
their state in parallel, according to the combinatorial logic of their
inputs, Kauffman's Random Boolean Networks \cite{K1}{K7}.
There is justified debate as to whether parallel (synchronous) updating,
and the on-off characterization of genes, are valid idealizations when applied
to real genomic networks, given that transcription is asynchronous and driven
at different rates. However, gene activity at the molecular scale consists of
discrete events occurring concurrently. Variable protein concentrations can be
accounted for by genes being on for some fraction of a given time span. The
RBN idealization is arguably a valid starting point for gaining insights into
gene network dynamics.
In a cell type's gene expression pattern over a span of time (i.e. its space-
time pattern), a particular gene may, broadly speaking, be either on, off, or
changing. If a large proportion of the genes are changing, chaotic dynamics, the cell will be unstable. On the other hand, dynamics that settles to
a pattern where a large proportion of the genes are permanently on or o
(frozen) may be too in exible for adaptive behavior. Cells constantly need to
adapt their gene expression pattern in response to a variety of hormone and
growth/di erentiation factors from nearby cells. The de nition of a cell type
may be more correctly expressed as a set of closely related gene expression
patterns, allowing an essential measure of exibility in behavior.
\section{Gene Regulatory Networks}

All steps of gene expression can be modulated, since passage of the transcription of DNA to RNA, to the post-translational modification of the protein
produced. Hence, gene expression is a complex process regulated at several stages in the synthesis of proteins. In addition to the DNA transcription regulation, the expression of a gene may be controlled during RNA processing and transport (in eukaryotes), RNA translation, and the post-translational modification of proteins. This gives rise to genetic regulatory systems structured by networks of regulatory interactions between DNA, RNA, proteins and other molecules [6]: a complex network termed as a \emph{gene regulatory
network} (GRN). Some kind of proteins are the transcription factors that bind to specific DNA sequences in order to regulate the
expression of a given gene. The power of transcription factors resides in their ability to activate and/or repress transcription of genes. The activation of
a gene is also referred to positive regulation, while the negative regulation
identifies the inhibition of the gene.
The regulation of gene expression is essential for the cell, because it
allows to control the internal and external functions of the cell. Furthermore,
in multicellular organisms, gene regulation drives the processes of cellular
differentiation and morphogenesis, leading to the creation of different cell
types that possess different gene expression profiles, and these last therefore
produce different proteins that have different ultrastructures that suit them
to their functions (though they all possess the genotype, which follows the
same genome sequence) 4. Therefore, with few exceptions, all cells in an
organism contain the same genetic material [6], and hence the same genome. The difference between the cells are emergent and due to regulatory mechanisms which can turn on or off genes. Two cells are different, if they have different subsets of active genes.
